src/app/clusters/closure-dimension-server/closure-dimension-cluster-logic.cpp

```diff
@@ -63,14 +64,22 @@ CHIP_ERROR ClusterLogic::Init(const ClusterConformance & conformance, const Clus
     return CHIP_NO_ERROR;
 }
 
-// TODO: CurrentState should be QuietReporting.
+// Specification rules for CurrentState quiet reporting:
+// Changes to this attribute SHALL only be marked as reportable in the following cases:
+// When the Position changes from null to any other value and vice versa, or
+// At most once every 5 seconds when the Position changes from one non-null value to another non-null value, or
+// When Target.Position is reached, or
+// When CurrentState.Speed changes, or
+// When CurrentState.Latch changes.
 CHIP_ERROR ClusterLogic::SetCurrentState(const DataModel::Nullable<GenericDimensionStateStruct> & incomingCurrentState)
```

Comment:
It seems to me that the spec is buggy here, in that changes from null to not-null for the entire attribute should trigger reporting, no?  Right now it doesn't look like they do, in the spec (but they do in the code below).

And given that they should do that, is there any value in using `QuieterReportingAttribute` to store this state?  I'm a little confused why we are using it for one of the fields in an odd way, not the whole struct....